### ISSUE TITLE
[subplat] Changes to Stripe to Firestore documents script

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#13687]
**Background**

After running the script in Stage we noticed some changes that are necessary for the script before moving to production.

**Requirements**

* Better naming of JSON files
  * Currently we only use the Product/Plan ID, making it very difficult to identify which file corresponds to which product/plan.
* Google API Auth
  * Use existing Google Auth config instead of setting new Environment Variable

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-5577)

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

{quote}Ignore archived products{quote}

Do we want to ignore archived products? A user can still have an active, ongoing subscription to an archived product. IIUC the VPN team plans to archive one of their production plans soon. We use some of this metadata config in other places than checkout like in recurring subscription emails, which we would still send to active subscribers of archived products.

I was actually coming here to request that we don’t hardcode active: true when inserting products/plans into Firestore (in stripeProductToProductConfig and stripePlanToPlanConfig).

**Comment by data-sync-user** [comment]
➤ Benjamin Bangert commented:

I do not plan on ignoring archived products. I’ll update the issue.

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

I would also recommend prettifying the JSON file output for better readability. This will be especially helpful once PMs/RPs start editing these files and opening PRs.

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

I also noticed that we only store valid product/plan configs in Firestore, but we create an empty directory for invalid products for the JSON output. This means I have 9 valid productConfigs in Firestore and 22 product directories for the JSON output in DEV. Is this desirable?

**Comment by data-sync-user** [comment]
➤ Barry Chen commented:

Well well well good thing I checked myself before I wrecked myself.  I was going to say that we should not ignore archived plans, since we’ll need the configs for them in our plans endpoint.  But, our plans endpoint currently only return active Stripe plans!  We still should not ignore the archived plans though; we’ll need the configs for them even though they no longer accept new subscriptions.

(The fact that we are only returning active Stripe plans from our plans endpoint can be an issue.  Maybe VPN has never archived a plan with active subscriptions?)

As for the “active” flag, we(?) decided(?) for it to mean whether the product/plan is accepting new subscriptions.  So it’s similar to the status property in Stripe product/prices, but a) not all config docs are for Stripe product/prices, and b) it’s more explicit as it’s returned from the plans endpoint (the “abbrev” plans do not include any status).  In other words, if the Stripe product/price is archived, let’s set “active” to false.

**Comment by data-sync-user** [comment]
➤ Reino Muhl commented:

For the archived Products/Plans that are currently failing Product/Plan Config validation checks, what would be the best/easiest approach to handling these cases?

  1. Should we update the Stripe metadata with some default values
  2. Should we ignore validation for archived products/plans?

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

{quote}For the archived Products/Plans that are currently failing Product/Plan Config validation checks, what would be the best/easiest approach to handling these cases?{quote}

Reino Muhl : I think it may depend on the environment, what the validation errors are, and if there are any active subscriptions. For example:

  * We probably don’t care about fixing dev products/plans with no active subscriptions.
  * We probably do care about fixing stage/prod products/plans with active subscriptions – particularly those created by an RP.
    * We wouldn’t want to use defaults here; if the validation errors are not straightforward fixes, we should ask the RP to fix it.

{quote}(The fact that we are only returning active Stripe plans from our plans endpoint can be an issue. Maybe VPN has never archived a plan with active subscriptions?){quote}

Thank you Barry Chen ; I filed https://mozilla-hub.atlassian.net/browse/FXA-5599 ( https://mozilla-hub.atlassian.net/browse/FXA-5599|smart-link ), and I pinged you in there in case I missed something.

{quote}I do not plan on ignoring archived products. I’ll update the issue.{quote}

Benjamin Bangert : I think we need to do a bit more than that and update stripeProductToProductConfig (source code ( https://github.com/mozilla/fxa/blob/0c31fbbd36818bf42107e8f95d9e13c7584c30fa/packages/fxa-auth-server/scripts/stripe-products-and-plans-to-firestore-documents/stripe-products-and-plans-converter.ts#L258 )) and stripePlanToPlanConfig (source code ( https://github.com/mozilla/fxa/blob/0c31fbbd36818bf42107e8f95d9e13c7584c30fa/packages/fxa-auth-server/scripts/stripe-products-and-plans-to-firestore-documents/stripe-products-and-plans-converter.ts#L318 )), so that it doesn’t hardcode active: true on any products/plans we’re processing in the script. Perhaps you are already planning on fixing that?
